### PR TITLE
Fix some stale references to value types

### PR DESCRIPTION
**Pull request by ngzhian**
Based on comments from https://github.com/WebAssembly/spec/pull/1391.

### CODE REVIEW

File: document/core/intro/overview.rst [hunk 1]
```diff
@@ -23,8 +23,8 @@ This language is structured around the following concepts.
   Instead, integers are interpreted by respective operations
   as either unsigned or signed in two’s complement representation.
 
-  In addition to the basic value types above, there is a single 128 bit wide
-  value type representing different types of packed data.
+  In addition to these basic number types above, there is a single 128 bit wide
```
Comment (rossberg) [inline document/core/intro/overview.rst:26]: Nit: remove "above", seems a bit redundant.

**Review by rossberg — APPROVED** [review]
Thanks, LGTM!

**Comment by ngzhian** [comment]
Not sure why build-core-spec is failing, it seems to be using the wrong katex version. Will investigate.

**Comment by ngzhian** [comment]
the submodule checkout isn't working right:

https://github.com/WebAssembly/simd/runs/4181145280?check_suite_focus=true#step:2:465

**Comment by ngzhian** [comment]
> the submodule checkout isn't working right:
> 
> https://github.com/WebAssembly/simd/runs/4181145280?check_suite_focus=true#step:2:465

My dumb mistake, I added some katex changes unintentionally.

**Pull request merged**